engine/src/main/java/org/hibernate/search/engine/reporting/IndexFailureContext.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.engine.reporting;
+
+import java.util.List;
+
+/**
+ * Contextual information about a failing index operation.
+ */
+public interface IndexFailureContext extends FailureContext {
+
+	/**
+	 * @return The list of index operations that weren't committed yet when the failure occurred.
+	 * Never {@code null}, but may be empty.
+	 * These operations may not have been applied to the index.
+	 * Use {@link Object#toString()} to get a textual representation of each operation.
+	 */
+	List<?> getUncommittedOperations();
+
```

Comment:
why do we not add a method to get the index name?

Reply:
Mostly because I can't get it easily in the case of Elasticsearch... I'll see what I can do.

Reply:
Created https://hibernate.atlassian.net/browse/HSEARCH-3741 , because it will be easier to address after some of the pending PRs have been merged.